app/ui/src/main/java/com/fsck/k9/activity/setup/AccountSetupBasics.java

```diff
@@ -346,16 +349,27 @@ private void onNext() {
             return;
         }
 
-        String email = mEmailView.getText().toString();
+        final String email = mEmailView.getText().toString();
         String[] emailParts = splitEmail(email);
         String domain = emailParts[1];
         mProvider = findProviderForDomain(domain);
         if (mProvider == null) {
-            /*
-             * We don't have default settings for this account, start the manual
-             * setup process.
-             */
-            onManualSetup();
+            new AccountDiscovery().discover(email, new Function1<AccountDiscovery.DiscoveredSettings, Unit>() {
```

Comment:
Whenever you find yourself invoking a constructor or static factory method inside a class, you're most likely not following proper dependency injection principles. We have lots of old code like this. But let's try to do better with new code.

Reply:
:+1: fixed with Koin for all introduced classes.